src/protobuf-net.Grpc/Internal/Reshape.cs

```diff
@@ -520,7 +520,10 @@ await foreach (var value in request.WithCancellation(allDone.Token).ConfigureAwa
             }
             catch
             {
-                allDone.Cancel();
+                if (!allDone.IsCancellationRequested)
```

Comment:
the fact that we need to test this suggests that there's competition, which suggests that this is itself an inherent race-condition; as such, I suggest putting a `try`/`catch` around the actual `Cancel()`; thoughts?